cdap-security/src/test/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementServiceTest.java

```diff
@@ -212,6 +216,30 @@ public void testAuthCacheFilter() throws Exception {
     }
   }
 
+  @Test
+  public void testResiliency() throws Exception {
+    CConfiguration cConfCopy = CConfiguration.copy(CCONF);
+    cConfCopy.setInt(Constants.Security.Authorization.CACHE_REFRESH_INTERVAL_SECS, 1);
+    DefaultAuthorizationEnforcementService authorizationEnforcementService =
+      new DefaultAuthorizationEnforcementService(new FailingPrivilegesFetcher(), cConfCopy);
+    Map<Principal, Map<EntityId, Set<Action>>> cache = authorizationEnforcementService.getCache();
+    cache.put(new Principal("bob", Principal.PrincipalType.USER), Collections.<EntityId, Set<Action>>emptyMap());
+    cache.put(new Principal("tom", Principal.PrincipalType.USER), Collections.<EntityId, Set<Action>>emptyMap());
+    authorizationEnforcementService.startAndWait();
+    try {
+      // allow the service to run for a few iterations - it should throw an exception every second
+      TimeUnit.SECONDS.sleep(2);
```

Comment:
Should we pass the `CountDownLatch` to the `FailingPrivilegesFetcher` and wait on it? So that it guarantees that the listPrivileges has executed once.